solutions/usaco-1061.mdx

```diff
@@ -106,4 +106,132 @@ for x in eat:
+				eastCows.add(new int[]{x, y, i});
+			}
+		}
+		/*
+		* https://docs.oracle.com/javase/8/docs/api/java/util/Comparator.html
+		* sort east cows by comparing x positions
+		* (second element in array)
+		*/
+		eastCows.sort(Comparator.comparingInt(o -> o[1]));
+		/*
+		 * sort east cows by comparing y positions
+		 * (first element in array)
+		 */
+		northCows.sort(Comparator.comparingInt(o -> o[0]));
+
+		// stores locations of stopped cows
+		int[] stoppedLocations = new int[n];
+		Arrays.fill(stoppedLocations, -1);
+		// check every combination of cows
+		for (int[] northCow : northCows) {
+			for (int[] eastCow : eastCows) {
+				/*
+				 * The north cows can't move down, and the east cows
+				 * can't move backwards, so a north cow's x coordinate must
+				 * be less than an east cow's x coordinate, and a north
+				 * cow's y coordinate must be less than an east cow's y coordinate
+				 * for them to collide.
+				  */
+				if (northCow[0] > eastCow[0] && northCow[1] < eastCow[1]) {
+					// Distance they travel
+					int northDistance = eastCow[1] - northCow[1];
+					int eastDistance = northCow[0] - eastCow[0];
+					/*
+					 * If the north cow gets to the meeting place before
+					 * the east cow and the east cow has not been stopped
+					 * already,
+					  */
+					if (northDistance < eastDistance && 
+						stoppedLocations[eastCow[2]] == -1) {
+						/*
+						 * then we can set the east cow's stopping point.
+						 * We only save the x coordinate because the y coordinate
+						 * stays constant.
+						 */
+						stoppedLocations[eastCow[2]] = northCow[0];
+					}
+
+					/*
+					 * If the east cow gets there before, 
+					 * we can save the north cow's position.
+					 */
+					if (northDistance > eastDistance && 
+						stoppedLocations[eastCow[2]] == -1) {
+						stoppedLocations[northCow[2]] = eastCow[1];
+						/*
+						 * We iterate per north cow, so we can move on to the
+						 * next north cow if this cow can't move anymore
+						 */
+						break;
+					}
+				}
+			}
+		}
+
+		// Tracks distance traveled.
+		int[] distances = new int[n];
+		Arrays.fill(distances, -1);
+		for (int[] northCow : northCows) {
+			/*
+			 * If this array element hasn't been touched, 
+			 * the cow never stops and eats forever.
+			 */
+			if (stoppedLocations[northCow[2]] != -1) {
+				// Eaten is (current position - original position) 
+				distances[northCow[2]] = stoppedLocations[northCow[2]] - northCow[1];
+			}
+
+		}
+
+		for (int[] eastCow : eastCows) {
+			if (stoppedLocations[eastCow[2]] != -1) {
+				distances[eastCow[2]] = stoppedLocations[eastCow[2]] - eastCow[0];
+			}
+		}
+
+		for (int x : distances) {
+			/* Ternary operator:
+			 * basically a condensed if/else statement
+			 * variable = (logical expression) ? [set to this value if true] : [set to this value if false];
+			 * https://docs.oracle.com/javase/tutorial/java/nutsandbolts/op2.html#:~:text=Another%20conditional%20operator,value2%20to%20result.%22
```

Comment:
this line is so long but you can't really split it since it's a url... maybe put the explanation of a ternary operator outside of the code and in an info block?